From f5003ef5b03bffd0df3adafc0bf81cfb2fbe33b8 Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Wed, 12 May 2021 14:56:19 +0200
Subject: [PATCH 06/24] stop looking for the plasma version in the session file

this was probably left over from when the module was outside plasma. now
that it is in kinfocenter, and kinfocenter is part of plasma, we can
simply use our own version as the plasma entry version. they are in sync
across all of plasma anyway and because of that kinfocenter's version is
as good an indication for the version of Plasma, the product, as the
session file installed by plasma-workspace

BUG: 436954
---
 Modules/about-distro/src/PlasmaEntry.cpp | 19 +++++--------------
 1 file changed, 5 insertions(+), 14 deletions(-)

diff --git a/Modules/about-distro/src/PlasmaEntry.cpp b/Modules/about-distro/src/PlasmaEntry.cpp
index 9e2fe55..6407ce3 100644
--- a/Modules/about-distro/src/PlasmaEntry.cpp
+++ b/Modules/about-distro/src/PlasmaEntry.cpp
@@ -1,5 +1,5 @@
 /*
-    SPDX-FileCopyrightText: 2012-2020 Harald Sitter <sitter@kde.org>
+    SPDX-FileCopyrightText: 2012-2021 Harald Sitter <sitter@kde.org>
     SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
 */
 
@@ -9,6 +9,8 @@
 #include <KConfigGroup>
 #include <KDesktopFile>
 
+#include "Version.h"
+
 PlasmaEntry::PlasmaEntry()
     : Entry(ki18n("KDE Plasma Version:"), plasmaVersion())
 {
@@ -21,17 +23,6 @@ PlasmaEntry::PlasmaEntry()
 
 QString PlasmaEntry::plasmaVersion()
 {
-    const QStringList &filePaths = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation,
-                                                             QStringLiteral("xsessions/plasma.desktop"));
-
-    if (filePaths.length() < 1) {
-        return QString();
-    }
-
-    // Despite the fact that there can be multiple desktop files we simply take
-    // the first one as users usually don't have xsessions/ in their $HOME
-    // data location, so the first match should (usually) be the only one and
-    // reflect the plasma session run.
-    KDesktopFile desktopFile(filePaths.first());
-    return desktopFile.desktopGroup().readEntry("X-KDE-PluginInfo-Version", QString());
+    // KInfoCenter is part of Plasma. Simply show our version!
+    return QString::fromLatin1(PROJECT_VERSION);
 }
-- 
2.31.1

